sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetQuerySuite.scala

```diff
@@ -1108,6 +1108,17 @@ abstract class ParquetQuerySuite extends QueryTest with ParquetTest with SharedS
       checkAnswer(sql("select * from tbl"), expected)
     }
   }
+
+  test("SPARK-39833: count() with pushed filters from Parquet files") {
```

Comment:
What if we change this test to:
```scala
  test("SPARK-39833-2: count() with pushed filters from Parquet files") {
    withTempPath { path =>
      val p = s"${path.getCanonicalPath}${File.separator}col=0${File.separator}"
      Seq((0, "a")).toDF("COL", "b").coalesce(1).write.save(p)
      val df = spark.read.parquet(path.getCanonicalPath)
      checkAnswer(df.filter("col = 0"), Seq(Row(0, "a")))
      assert(df.filter("col = 0").select('b).collect().toSeq == Row("a") :: Nil)
    }
  }
```

it seems checking whether `requestedSchema` is empty is not sufficient enough, since it can be non-empty but the pushed filter could reference column that do not exist in it.

Reply:
Yes, the provided test fails. it appears the problem is wide-spread due to this parquet-mr bug. I would have to change the code to check filters that are pushed down against the requested schema instead of a full one but then it might introduce a performance issue.

I think the safest for now is to disable column indexes altogether. I will take a look at what could be done. 
We can also merge this PR and I can follow up on the fix later.

Reply:
>  ... I would have to change the code to check filters that are pushed down against the requested schema instead of a full one but then it might introduce a performance issue.

Yes I think this would be the approach, but why it might introduce performance issues?

Reply:
There will be no predicate pushdown in this case so reads could be slower potentially.

Reply:
I think we may need to disable column index altogether until we figure out the proper fix. I verified that `hadoopConf.setBoolean(ParquetInputFormat.COLUMN_INDEX_FILTERING_ENABLED, false)` makes both tests work.

Reply:
Actually, there seems to be another bug in DSv2 which I will fix later: 
```
checkAnswer(df.filter("col = 0"), Seq(Row(0, "a")))
```
fails in DSv2 due to a different column order.

Let's do one problem at a time.